test/integration/kubernetes_deploy_test.rb

```diff
@@ -1,27 +1,134 @@
+
+  def test_invalid_yaml_fails_fast
+    error = assert_raises(KubernetesDeploy::FatalDeploymentError) do
+      deploy_fixture_set("invalid", subset: ["yaml-error"])
+    end
+    assert_match /Template \S+ cannot be parsed/, error.to_s
+  end
+
+  def test_invalid_k8s_spec_that_is_valid_yaml_fails_fast
+    fixture_set = load_fixture_data("basic", ["configmap-data"])
+    configmap = fixture_set["configmap-data"]["ConfigMap"].first
+    configmap["metadata"]["myKey"] = "uhOh"
+
+    error = assert_raises(KubernetesDeploy::FatalDeploymentError) do
+      deploy_loaded_fixture_set(fixture_set)
+    end
+    assert_match /Dry run failed for template configmap-data/, error.to_s
+
+    @logger_stream.rewind
+    assert_match /error validating data\: found invalid field myKey for v1.ObjectMeta/, @logger_stream.read
+  end
+
+  def test_dead_pods_in_old_replicaset_are_ignored
+    fixture_set = load_fixture_data("basic", ["configmap-data", "web"])
+    deployment = fixture_set["web"]["Deployment"].first
+    container = deployment["spec"]["template"]["spec"]["activeDeadlineSeconds"] = 1
+    deploy_loaded_fixture_set(fixture_set, wait: false) # this will never succeed as pods are killed after 1s
+
+    sleep 1 # make sure to hit DeadlineExceeded on at least one pod
+
+    deploy_fixture_set("basic", subset: ["web", "configmap-data"])
+    pods = kubeclient.get_pods(namespace: @namespace, label_selector: "name=web,app=basic")
+    running_pods, not_running_pods = pods.partition { |pod| pod.status.phase == "Running" }
+    assert_equal 1, running_pods.size
+    assert not_running_pods.size >= 1
+  end
+
+  def test_bad_container_image_on_run_once_halts_and_fails_deploy
+    fixture_set = load_fixture_data("basic")
+    pod = fixture_set["unmanaged-pod"]["Pod"].first
+    pod["spec"]["activeDeadlineSeconds"] = 3
+    pod["spec"]["containers"].first["image"] = "hello-world:elephants"
+
+    error = assert_raises(KubernetesDeploy::FatalDeploymentError) do
+      deploy_loaded_fixture_set(fixture_set)
     end
+    assert_match /1 priority resources failed to deploy/, error.to_s
+
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_unmanaged_pod_statuses("Failed")
+    basic.assert_configmap_data_present # priority resource
+    basic.refute_redis_resources_exist(expect_pvc: true) # pvc is priority resource
+    basic.refute_web_resources_exist
+  end
+
+  def test_wait_false_still_waits_for_priority_resources
+    fixture_set = load_fixture_data("basic")
+    pod = fixture_set["unmanaged-pod"]["Pod"].first
+    pod["spec"]["activeDeadlineSeconds"] = 1
+    pod["spec"]["containers"].first["image"] = "hello-world:elephants"
 
-    assert_equal 2, managed_pods.size
-    managed_pods.each do |pod|
-      assert_equal "Running", pod.status.phase
+    error = assert_raises(KubernetesDeploy::FatalDeploymentError) do
+      deploy_loaded_fixture_set(fixture_set)
     end
+    assert_match /1 priority resources failed to deploy/, error.to_s
```

Comment:
Do we want to assert that something about the image not being found is in the logging output?